layouts/solo/loader.html

```diff
@@ -0,0 +1,144 @@
+                    if (!immediate) func.apply(context, args);
+                };
+                var callNow = immediate && !timeout;
+                clearTimeout(timeout);
+                timeout = setTimeout(later, wait);
+                if (callNow) func.apply(context, args);
+            };
+        };
+
+        var extend = function(o) {
+            o = o || {};
+            for (var i = 1; i < arguments.length; i++) {
+                if (!arguments[i])
+                    continue;
+                for (var key in arguments[i]) {
+                    if (arguments[i].hasOwnProperty(key))
+                        o[key] = arguments[i][key];
+                }
+            }
+            return o;
+        };
+
+        //Dynamically creates an iframe & appends it's required CSS & JS libraries.
+        var getIframeHtml = function(options) {
+            var html = '<head><base target="_blank" /></head><body class="' + (options.className || '') + '" data-domain="' +
+                (options.domain || '') + '" data-id="' + (options.id || '') + '" onload="' +
+                'var d = document, head = d.getElementsByTagName(\'head\')[0],' +
+                'addJS = function(u){ var s = d.createElement(\'script\');s.src=u;d.body.appendChild(s);},' +
+                'addCSS = function(h){ var l = d.createElement(\'link\');l.type=\'text/css\';l.rel=\'stylesheet\';l.href=h;head.appendChild(l);};' +
+                'window.DEBUG=' + DEBUG + ';';
+
+            var js = options.js || [];
+            if ('function' === typeof js) {
+                js = js();
+            }
+
+            js = js.filter(Boolean);
+
+            for (var i = 0; i < js.length; i++) {
+                html += 'addJS(\'' + js[i] + '\');';
+            }
+
+            var css = options.css || [];
+            if ('function' === typeof css) {
+                css = css();
+            }
+
+            css = css.filter(Boolean);
+            for (var i = 0; i < css.length; i++) {
+                html += 'addCSS(\'' + css[i] + '\');';
+            }
+
+            html += '"><style>' + (options.style || '') + '</style>';
+            var content = options.html || '';
+            if ('function' === typeof content) {
+                html += content();
+            } else if (content.trim().length) {
+                html += content;
+            }
+
+            return html;
+        };
+
+        window.__TVPage__ = window.__TVPage__ || {};
+        __TVPage__.config = __TVPage__.config || {};
+        __TVPage__.config["{{ $id }}"] = extend({{ $.Scratch.Get "settings" }}, __TVPage__.config["{{ $id }}"]);
+
+        var holder = document.getElementById("{{ $id }}-holder");
+        var iframe = holder.querySelector(".tvp-iframe");
+        var config = __TVPage__.config["{{ $id }}"];
+        var staticPath = "{{ .Type | absURL }}";
+
+        var dataMethod = "static";
+        if (isset(config, "channel") && isset(config.channel, "id")) {
```

Comment:
Yes, I'm considering that a widget uses a "dynamic data method" as long as a channel object (standard or programmatic) is being passed. If a widget has to fetch its data and build itself on the fly, then it fits in this category.

Please let me know if you consider this is robust.